### PR TITLE
[Scala 3] Properly show enum on hover

### PR DESCRIPTION
**Pull request by tgodzik**
This was actually done badly* previously and not actual tests were present, which is why this issue sneaked trough

*Done badly by me :sweat_smile:

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala [hunk 1]
```diff
@@ -40,12 +40,12 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
...
         } yield name
         "package " + owners.toList.reverse.mkString(".")
-      // enum
-      case _ if sym.companionClass.is(Flags.Enum) =>
-        s"enum $name: "
       // enum case
       case _ if sym.is(Flags.EnumVal) =>
         s"case $name: "
+      // enum
+      case _ if sym.is(Flags.Enum) =>
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:47]: `EnumVal` is also an enum, which is why I switched the order around.

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala [hunk 2]
```diff
@@ -73,7 +73,12 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
     val typeSymbol = info.typeSymbol
     sym match {
       case p if p.is(Flags.Package) => fullDef
-      case p if typeSymbol.is(Flags.Module) =>
+      /* Type cannot be shown on the right since it is already a type
+       * let's instead use that space to show the full path.
+       */
+      case p
+          if typeSymbol.isOneOf(Flags.Module) || // object
```
Comment (dos65) [inline mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:80]:
```suggestion
          if typeSymbol.is(Flags.Module) || // object
```

**Review by dos65 — APPROVED** [review]
Cool!
I've seen this issue but thought that the reason should be more close to compiler :smile:

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 2 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala]: Fixed!

**Pull request merged**